armi/reactor/blocks.py

```diff
@@ -796,9 +795,16 @@ def completeInitialLoading(self, bolBlock=None):
         for child in self:
             hmMass = child.getHMMass() * sf
             massHmBOL += hmMass
-            # Components have a massHmBOL parameter but not every composite will
+            # Components have the following parameters but not every composite will
+            # massHmBOL, molesHmBOL, puFrac
             if isinstance(child, components.Component):
                 child.p.massHmBOL = hmMass
+                child.p.molesHmBOL = child.getHMMoles()
```

Comment:
I see we are modifying `Block.completeInitialLoading()`, but I don't see a test for the change.  Thoughts?

Reply:
https://github.com/terrapower/armi/pull/1990/commits/8c94b3200134732ce306bd43828b2fd9b4e8cd98

that was more involved than I thought. I do think we need to think carefully about symmetry factors for components.